tests/trainer/connectors/test_checkpoint_connector.py

```diff
@@ -126,3 +130,51 @@ def test_hpc_max_ckpt_version(tmpdir):
     assert trainer.checkpoint_connector.hpc_resume_path == str(tmpdir / "hpc_ckpt_33.ckpt")
     assert trainer.checkpoint_connector.max_ckpt_version_in_folder(tmpdir) == 33
     assert trainer.checkpoint_connector.max_ckpt_version_in_folder(tmpdir / "not" / "existing") is None
+
+
+@mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": "1"})
+@RunIf(min_torch="1.7.0")
+def test_loops_restore(tmpdir):
+    """Test that required loop state_dict is loaded correctly by checkpoint connector."""
+    model = BoringModel()
+    checkpoint_callback = ModelCheckpoint(dirpath=tmpdir, save_last=True)
+    trainer_args = dict(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        limit_train_batches=2,
+        limit_val_batches=2,
+        logger=False,
+        callbacks=[checkpoint_callback],
+        num_sanity_val_steps=0,
+    )
+    trainer = Trainer(**trainer_args)
+    trainer.validate(model)
+    trainer.test(model)
+    trainer.predict(model)
+    trainer.fit(model)
```

Comment:
Not a fan of this test as it tries to unit-test the `load_state_dict` behaviour but still runs the trainer for all these entry-points

I would either remove it or reduce the scope so it interacts with the connector directly without using the trainer